x-pack/test/functional/services/ml/trained_models_table.ts

```diff
@@ -226,6 +239,256 @@ export function TrainedModelsTableProvider(
+      tag: string;
+    }) {
+      // INFERENCE CONFIG
+      const editInferenceConfigButton = await testSubjects.find(
+        'mlTrainedModelsInferencePipelineInferenceConfigEditButton'
+      );
+      await editInferenceConfigButton.click();
+      await this.setTrainedModelsInferenceFlyoutCustomEditorValues(
+        'mlTrainedModelsInferencePipelineInferenceConfigEditor',
+        JSON.stringify(values.editedInferenceConfig)
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+      // FIELD MAP
+      const editFieldMapButton = await testSubjects.find(
+        'mlTrainedModelsInferencePipelineFieldMapEditButton'
+      );
+      await editFieldMapButton.click();
+      await this.setTrainedModelsInferenceFlyoutCustomEditorValues(
+        'mlTrainedModelsInferencePipelineFieldMapEdit',
+        JSON.stringify(values.editedFieldMap)
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+      // OPEN ADVANCED SETTINGS
+      await this.trainedModelsInferenceOpenAdditionalSettings();
+      await this.setTrainedModelsInferenceFlyoutCustomValue(
+        'mlTrainedModelsInferenceAdvancedSettingsConditionTextArea',
+        values.condition
+      );
+      await this.setTrainedModelsInferenceFlyoutCustomValue(
+        'mlTrainedModelsInferenceAdvancedSettingsTagInput',
+        values.tag
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+
+      await this.assertTrainedModelsInferenceFlyoutPipelineConfigValues(
+        values.editedInferenceConfig,
+        values.editedFieldMap
+      );
+      await this.assertTrainedModelsInferenceFlyoutAdditionalSettings(values.condition, values.tag);
+    }
+
+    public async completeTrainedModelsInferenceFlyoutDetails(
+      expectedValues: {
+        name: string;
+        description: string;
+        targetField: string;
+      },
+      editDefaults: boolean = false
+    ) {
+      if (editDefaults) {
+        await this.setTrainedModelsInferenceFlyoutCustomDetails(expectedValues);
+      }
+      const name = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineNameInput',
+        'value'
+      );
+      expect(name).to.eql(expectedValues.name);
+      const description = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineDescriptionInput',
+        'value'
+      );
+      expect(description).to.eql(expectedValues.description);
+      const targetField = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineTargetFieldInput',
+        'value'
+      );
+      expect(targetField).to.eql(expectedValues.targetField);
+      await this.deployModelsContinue('mlTrainedModelsInferencePipelineProcessorConfigStep');
+    }
+
+    public async trainedModelsInferenceOpenAdditionalSettings() {
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsAccordion');
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsAccordion');
```

Comment:
duplicated assertion 
```suggestion
      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsAccordion');
```